### PR TITLE
Remove legacy targets

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - depends on https://github.com/openvinotoolkit/openvino/pull/9298

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: cmake/developer_package/add_ie_target.cmake [hunk 1]
```diff
@@ -133,6 +133,10 @@ function(addIeTarget)
     endif()
 endfunction()
 
+function(add_ov_target)
+    addIeTarget(${ARGV})
+endfunction()
```
Comment (ilya-lavrenov) [inline cmake/developer_package/add_ie_target.cmake:138]: `ov_` prefix should be at the beginning I suppose

File: cmake/extra_modules.cmake [hunk 2]
```diff
@@ -63,22 +63,32 @@ function(register_extra_modules)
     openvino_developer_export_targets(COMPONENT core TARGETS ngraph)
 
     set(InferenceEngineDeveloperPackage_DIR "${CMAKE_CURRENT_BINARY_DIR}/runtime")
+    set(OpenVINODeveloperPackage_DIR "${CMAKE_BINARY_DIR}/runtime")
 
-    function(generate_fake_dev_package)
-        set(iedevconfig_file "${InferenceEngineDeveloperPackage_DIR}/InferenceEngineDeveloperPackageConfig.cmake")
-        file(REMOVE "${iedevconfig_file}")
+    function(generate_fake_dev_package NS)
+        if ("${NS}" STREQUAL "openvino")
```
Comment (ilya-lavrenov) [inline cmake/extra_modules.cmake:69]: `"${NS}"` -> `NS`

File: cmake/extra_modules.cmake [hunk 3]
```diff
@@ -63,22 +63,32 @@ function(register_extra_modules)
...
         foreach(target IN LISTS ${openvino_export_components})
             if(target)
-                file(APPEND "${iedevconfig_file}" "add_library(IE::${target} ALIAS ${target})\n")
+                file(APPEND "${devconfig_file}" "add_library(${NS}::${target} ALIAS ${target})\n")
             endif()
         endforeach()
+        if ("${NS}" STREQUAL "openvino")
+            file(APPEND "${devconfig_file}" "add_library(${NS}::runtime ALIAS ov_runtime)\n")
+            file(APPEND "${devconfig_file}" "add_library(${NS}::runtime::dev ALIAS ov_runtime_dev)\n")
+        endif()
```
Comment (ilya-lavrenov) [inline cmake/extra_modules.cmake:87]: why do we need to add them explicitly?

File: docs/template_plugin/tests/functional/CMakeLists.txt [hunk 4]
```diff
@@ -5,13 +5,13 @@
 # [cmake:functional_tests]
 set(TARGET_NAME templateFuncTests)
 
-addIeTargetTest(
+add_ov_test_target(
```
Comment (ilya-lavrenov) [inline docs/template_plugin/tests/functional/CMakeLists.txt:8]: `add_ov_test_target` => `ov_add_test_target`

File: src/common/legacy/CMakeLists.txt [hunk 5]
```diff
@@ -37,9 +37,9 @@ target_include_directories(${TARGET_NAME}_obj PRIVATE
     ${PUBLIC_HEADERS_DIR}
     ${CMAKE_CURRENT_SOURCE_DIR}/src
     $<TARGET_PROPERTY:inference_engine_obj,SOURCE_DIR>/src # For CNNNetworkNGraphImpl
-    $<TARGET_PROPERTY:inference_engine_transformations,INTERFACE_INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:openvino::runtime::dev,INTERFACE_INCLUDE_DIRECTORIES>
     $<TARGET_PROPERTY:inference_engine_plugin_api,INTERFACE_INCLUDE_DIRECTORIES>
```
Comment (ilya-lavrenov) [inline src/common/legacy/CMakeLists.txt:41]: do we need `inference_engine_plugin_api`? I suppose `openvino::runtime::dev` should replace it in all the places

File: src/common/legacy/CMakeLists.txt [hunk 6]
```diff
@@ -37,9 +37,9 @@ target_include_directories(${TARGET_NAME}_obj PRIVATE
...
-    $<TARGET_PROPERTY:ngraph,INTERFACE_INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:ngraph_obj,INTERFACE_INCLUDE_DIRECTORIES>
```
Comment (ilya-lavrenov) [inline src/common/legacy/CMakeLists.txt:42]: I suppose `openvino::runtime::dev` already provide all public include headers

File: src/common/offline_transformations/CMakeLists.txt [hunk 7]
```diff
@@ -19,8 +19,8 @@ source_group("include" FILES ${PUBLIC_HEADERS})
 
 add_library(${TARGET_NAME} STATIC EXCLUDE_FROM_ALL ${LIBRARY_SRC} ${PUBLIC_HEADERS})
 
-target_link_libraries(${TARGET_NAME} PUBLIC ngraph inference_engine_transformations ngraph::reference
-                                     PRIVATE openvino::itt pugixml::static openvino::frontend::common
+target_link_libraries(${TARGET_NAME} PUBLIC openvino::runtime::dev ngraph::reference
```
Comment (ilya-lavrenov) [inline src/common/offline_transformations/CMakeLists.txt:22]: why `ngraph::reference` is `PUBLIC`?

File: src/common/preprocessing/CMakeLists.txt [hunk 8]
```diff
@@ -99,7 +99,7 @@ ie_faster_build(${TARGET_NAME}_obj
 )
 
 target_compile_definitions(${TARGET_NAME}_obj PRIVATE IMPLEMENT_INFERENCE_ENGINE_PLUGIN
-                                                      $<TARGET_PROPERTY:inference_engine,INTERFACE_COMPILE_DEFINITIONS>
+                                                      $<TARGET_PROPERTY:ov_runtime,INTERFACE_COMPILE_DEFINITIONS>
```
Comment (ilya-lavrenov) [inline src/common/preprocessing/CMakeLists.txt:102]: should we use `openvino::runtime`?

File: src/core/tests/engines_util/CMakeLists.txt [hunk 9]
```diff
@@ -8,9 +8,9 @@ add_library(engines_test_util STATIC EXCLUDE_FROM_ALL ${ENGINES_UTIL_SRC})
 
 ie_faster_build(engines_test_util UNITY)
 
-target_include_directories(engines_test_util PRIVATE $<TARGET_PROPERTY:inference_engine,INTERFACE_INCLUDE_DIRECTORIES>)
-target_include_directories(engines_test_util PRIVATE $<TARGET_PROPERTY:inference_engine_transformations,INTERFACE_INCLUDE_DIRECTORIES>)
-target_link_libraries(engines_test_util PUBLIC ngraph inference_engine inference_engine_transformations gtest gmock ngraph_test_util)
+target_include_directories(engines_test_util PRIVATE $<TARGET_PROPERTY:openvino::runtime,INTERFACE_INCLUDE_DIRECTORIES>)
+target_include_directories(engines_test_util PRIVATE $<TARGET_PROPERTY:openvino::runtime::dev,INTERFACE_INCLUDE_DIRECTORIES>)
```
Comment (ilya-lavrenov) [inline src/core/tests/engines_util/CMakeLists.txt:12]: Why do we need `target_include_directories` while we have `target_link_libraries` below?

File: src/frontends/onnx/frontend/CMakeLists.txt [hunk 10]
```diff
@@ -7,7 +7,7 @@ ov_add_frontend(NAME onnx
                 PROTOBUF_LITE
                 SKIP_NCC_STYLE
                 FILEDESCRIPTION "FrontEnd to load and convert ONNX file format"
-                LINK_LIBRARIES ngraph::builder openvino::util onnx_common inference_engine_transformations)
+                LINK_LIBRARIES ngraph::builder openvino::util onnx_common openvino::runtime::dev)
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/CMakeLists.txt:10]: why do we need `dev`?

File: src/frontends/ir/CMakeLists.txt [hunk 11]
```diff
@@ -4,9 +4,8 @@
 
 ov_add_frontend(NAME ir
                 FILEDESCRIPTION "FrontEnd to load OpenVINO IR file format"
-                LINK_LIBRARIES inference_engine_transformations pugixml::static
-                               # TODO: remove dependencies below in CVS-69781
-                               inference_engine inference_engine_plugin_api)
+                LINK_LIBRARIES pugixml::static
+                               openvino::runtime::dev)
```
Comment (ilya-lavrenov) [inline src/frontends/ir/CMakeLists.txt:8]: I suppose we need to keep comment and once it's applied, we need `openvino::runtime::dev` -> `openvino::runtime`

File: src/frontends/paddlepaddle/CMakeLists.txt [hunk 12]
```diff
@@ -6,4 +6,4 @@
 ov_add_frontend(NAME paddlepaddle
                 PROTOBUF_LITE
                 FILEDESCRIPTION "FrontEnd to load and convert PaddlePaddle file format"
-                LINK_LIBRARIES inference_engine_transformations)
+                LINK_LIBRARIES openvino::runtime::dev)
```
Comment (ilya-lavrenov) [inline src/frontends/paddlepaddle/CMakeLists.txt:9]: why `dev`?

File: src/plugins/intel_gpu/CMakeLists.txt [hunk 13]
```diff
@@ -29,14 +29,11 @@ target_compile_options(${TARGET_NAME} PRIVATE
...
-                                             pugixml::static
-                                             inference_engine_transformations
-                                             inference_engine_lp_transformations
-                                             ngraph)
+                                             pugixml::static)
 
 target_include_directories(${TARGET_NAME} PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}/include/
-        $<TARGET_PROPERTY:inference_engine_transformations,INTERFACE_INCLUDE_DIRECTORIES>)
+        $<TARGET_PROPERTY:openvino::runtime::dev,INTERFACE_INCLUDE_DIRECTORIES>)
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/CMakeLists.txt:36]: do we need `openvino::runtime::dev` here? It's already part of `ov_add_plugin`

File: src/tests/functional/plugin/conformance/subgraphs_dumper/tests/CMakeLists.txt [hunk 14]
```diff
@@ -10,11 +10,11 @@ addIeTargetTest(
         ADDITIONAL_SOURCE_DIRS ${OpenVINO_SOURCE_DIR}/src/tests/functional/plugin/conformance/subgraphs_dumper/src/matchers
         INCLUDES
             ${OpenVINO_SOURCE_DIR}/src/tests/functional/plugin/conformance/subgraphs_dumper/include
-            $<TARGET_PROPERTY:inference_engine,INTERFACE_INCLUDE_DIRECTORIES>
+            $<TARGET_PROPERTY:openvino::runtime,INTERFACE_INCLUDE_DIRECTORIES>
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/tests/CMakeLists.txt:13]: we have link private below, suppose explicit includes are not needed

File: src/frontends/onnx/frontend/CMakeLists.txt (hunk 10 above)
Comment (ilyachur) [reply src/frontends/onnx/frontend/CMakeLists.txt:10]: dev contains transformation API

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**